.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,340 @@
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Spotbugs
+        run: |
+          ./gradlew spotbugsMain
+          ./gradlew spotbugsTest
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        run: cp -r ./service/build/reports ./artifacts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Spotbugs Reports'
+          path: ./artifacts/**
+          compression-level: 9
+
+      - name: Delete Temporary Artifacts
+        run: rm -r ./artifacts
+
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    needs: [checkstyle, spotbugs]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Execute Unit Tests
+        run: ./gradlew test
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        run: cp -r ./service/build/reports ./artifacts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Unit Test Reports'
+          path: ./artifacts/**
+          compression-level: 9
+
+      - name: Delete Temporary Artifacts
+        run: rm -r ./artifacts
+
+  integration-tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    needs: [checkstyle, spotbugs]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Start Docker Dependencies
+        env:
+          GPC_SUPPLIER_ODS_CODE: "XYZ222"
+          GP2GP_SERVER_PORT: "8080"
+          GP2GP_AMQP_BROKERS: "amqp://activemq:5672"
+          GP2GP_MONGO_URI: "mongodb://mongodb:27017"
+          GP2GP_MONGO_DATABASE_NAME: "gp2gp"
+          GP2GP_MHS_OUTBOUND_URL: "http://mock-mhs-adaptor:8081/mock-mhs-endpoint/accept-only"
+          GP2GP_GPC_GET_URL: "http://gpcc:8090/@ODS_CODE@/STU3/1/gpconnect"
+          GP2GP_LARGE_ATTACHMENT_THRESHOLD: "4500000"
+          GP2GP_LARGE_EHR_EXTRACT_THRESHOLD: "4500000"
+          GPC_CONSUMER_SERVER_PORT: "8090"
+          GPC_CONSUMER_SDS_URL: "http://wiremock:8080/spine-directory/"
+          GPC_CONSUMER_SDS_APIKEY: "anykey"
+          GP2GP_LOGGING_LEVEL: DEBUG
+          GPC_CONSUMER_LOGGING_LEVEL: DEBUG
+        run: |
+          docker network create commonforgp2gp
```

Comment:
The command `docker network create commonforgp2gp` will fail if the network already exists. You could add a check to ensure the network only gets created if it doesn't already exist:
`docker network inspect commonforgp2gp >/dev/null 2>&1 || docker network create commonforgp2gp`

Reply:
Hey @ole4ryb - great point, though given the way that the workflow is set-up I can't see any points in which it would be _possible_ to fail, given that the network `commonforgp2gp` exists.

Reply:
just thinking more from a re-run point of view, when something fails, not sure how GitHub Actions behave it that case and whether it preserves the docker network or destroys it... probably something to explore...

Reply:
So AFAIK:

* If a job fails it wouldn't use the same runner.
* As a failsafe, we have implemented:

```yml
- name: Stop Docker Dependencies
  if: always()
  run: |
    docker compose -f docker-compose.yml -f docker-compose-e2e-tests.yml down
    docker network rm commonforgp2gp
  working-directory: ./docker
```

The `if: always()` in the above example ensures that this step is always run, despite a failure - so the `commonforgp2gp` network will always be removed.

Reply:
sure, not a problem, I am leaving it completely up to you to assess the level of the risk.